cpp/tensorrt_llm/batch_manager/dataTransceiver.cpp

```diff
@@ -804,7 +820,15 @@ class CacheReceiver::Impl
         TLLM_CHECK_WITH_INFO(mFormatter->inquireSupport(mSelfState.getCacheState().value(), destCacheState),
             "Disagg server does not currently support these cacheState.");
 
-        RequestInfo requestInfo(requestId, mSelfState);
+        auto endpoint = contextState.getUniqueIdServerEndpoint();
+        TLLM_CHECK_WITH_INFO(endpoint.has_value(), "UniqueIdServer endpoint not found in DataTransceiverState");
+        auto const& parallelConfig = mSelfState.getCacheState().value().getParallelConfig();
+        auto const expectedRefCount = parallelConfig.mTensorParallelism * parallelConfig.mPipelineParallelism
+            * parallelConfig.mContextParallelism;
+        int32_t uniqueId = UniqueIdClient::instance().getUniqueId(
+            endpoint.value(), llmRequest.mRequestId, mServerUuid, expectedRefCount);
```

Comment:
I would explicitely have a variable `receiverTransferId = llmRequest.mRequestId`. Eventually, we might not have an `llmRequest` and we would need to generate `receiverTransferId` on the receiver side. This could get incremented for every new "generation_only" request, or for any new request to receive arbitrary KV cache blocks...